api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -320,8 +320,13 @@ private boolean checkWorkspaceWriteAccess(String workspaceNamespace, String work
               workspaceId.getWorkspaceName(),
               e.getResponseBody()),
           e);
-      // TODO: figure out what happens if the workspace already exists
-      throw new ServerErrorException("Error creating FC workspace", e);
+      if (e.getCode() == 403) {
```

Comment:
One thing missing from this: we have logic in generateFirecloudWorkspaceId that generates an FC name from a human readable name. It's possible for the human-readable names to differ while the generated names collide. 

I think probably what we should do is 1) have uniqueness checking of human-readable names in here, and throw a 409 if the name matches; and 2) update generateFirecloudWorkspaceId() so that it tacks on some numeric suffix or something to generate another FC name for the workspace when there's an FC name conflict, as we do for billing projects here:

https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/api/ProfileController.java#L133

Basically, the user should just have to worry about making the display names unique; they shouldn't have to care about what the workspace name is in FC much.

Reply:
Somewhat related: we should probably enforce uniqueness of display names with a unique index in the database, and stop making description on workspaces be required. I had a PR to do that which I'll send out soonish.